Update hdf5 to 1.10.5, netcdf to 4.6.3

bebosudo opened the pull request:
NetCDF 4.6.3 announcement:
https://www.unidata.ucar.edu/mailing_lists/archives/netcdfgroup/2019/msg00020.html

HDF5 1.10.5 announcement:
https://www.hdfgroup.org/2019/02/hdf5-1-10-5-release-newsletter-168/

michaelkuhn commented on var/spack/repos/builtin/packages/netcdf/package.py:
```diff
@@ -23,11 +23,11 @@ class Netcdf(AutotoolsPackage):
-            url="https://www.gfd-dennou.org/arch/netcdf/unidata-mirror/netcdf-c-4.6.2.tar.gz")
+    version('4.6.3',   sha256='335fdf16d7531f430ad75e732ed1a9a3fc83ad3ef91fb33a70119a555dd5415c')
+    version('4.6.2',   sha256='c37525981167b3cd82d32e1afa3022afb94e59287db5f116c57f5ed4d9c6a638')
     version('4.6.1',   sha256='89c7957458740b763ae828c345240b8a1d29c2c1fed0f065f99b73181b0b2642')
```
4.6.2 and 4.6.3 cannot be fetched with your changes (see #10786).

bebosudo commented:
You are right @michaelkuhn, I've added the specific URL also for the latest version, until #10786 is fixed.

gsjaardema commented:
Similar NetCDF change in #10786.  I added a `url_for+version` function in that patch to deal with the name change in 4.6.2 and following (`netcdf` to `netcdf-c`).  There is also #10788 which fixes a bad dependency on hdf5 which results in a `~mpi` installation depending on `hdf5+mpi` instead of `hdf5~mpi`

bebosudo commented:
Thanks for your fixes @gsjaardema: I did notice #10788 before, and solved by manually specifing `hdf5~mpi`.
Should we do something? apart from waiting for one of the PR to be accepted?

gsjaardema commented:
@bebosudo Probably just wait.  It has been that way for a long time, so waiting a short time shouldn't matter.

If you do `spack checksum hdf5` does it show all the versions for you?  I am unable to get it to work on my system (which is behind a firewall, but other packages work ok...).  I know the HDF5 group restructured some of their web sites, so not sure if that is the problem or if problem is on my side.  

I know it worked on previous version though...

bebosudo commented:
It doesn't seem to be deterministic: on the first run it found:
```
==> Found 64 versions of hdf5:
  
  5_1_10_Examples              https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.10/CMake/HDF5_1_10_Examples.tar.gz
  1.10.5                       https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.10/hdf5-1.10.5/src/hdf5-1.10.5.tar.gz
  1.10.2                       https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.10/hdf5-1.10.2/src/hdf5-1.10.2.tar.gz
  1.10.1                       https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.10/hdf5-1.10.1/src/hdf5-1.10.1.tar.gz
  1.10.0                       https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.10/hdf5-1.10.0/src/hdf5-1.10.0.tar.gz
  1.8.21                       https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.8/hdf5-1.8.21/src/hdf5-1.8.21.tar.gz
  1.8.20                       https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.8/hdf5-1.8.20/src/hdf5-1.8.20.tar.gz
  1.8.18                       https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.8/hdf5-1.8.18/src/hdf5-1.8.18.tar.gz
  1.8.17                       https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.8/hdf5-1.8.17/src/hdf5-1.8.17.tar.gz
  ...
  0.2.1                        https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.8/hdf5-1.8.18/src/HDF5Examples-0.2.1-Source.tar.gz
```

On the second:
```
==> Found 35 versions of hdf5:
  
  5-180-VMS                        https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.8/hdf5-1.8.0/src/HDF5-180-VMS.tar.gz
  1.10.1                           https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.10/hdf5-1.10.1/src/hdf5-1.10.1.tar.gz
  1.8.18                           https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.8/hdf5-1.8.18/src/hdf5-1.8.18.tar.gz
  1.8.11-linux-shared              https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.8/hdf5-1.8.11/bin/linux/hdf5-1.8.11-linux-shared.tar.gz
  1.8.11                           https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.8/hdf5-1.8.11/src/hdf5-1.8.11.tar.gz
  1.8.7                            https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.8/hdf5-1.8.7/src/hdf5-1.8.7.tar.gz
  1.8.6                            https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.8/hdf5-1.8.6/src/hdf5-1.8.6.tar.gz
  1.8.4-patch1-solaris-static      https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.8/hdf5-1.8.4-patch1/bin/solaris/hdf5-1.8.4-patch1-solaris-static.tar.gz
  1.8.4-patch1-solaris-shared      https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.8/hdf5-1.8.4-patch1/bin/solaris/hdf5-1.8.4-patch1-solaris-shared.tar.gz
  ...
  0.2.1                            https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.8/hdf5-1.8.18/src/HDF5Examples-0.2.1-Source.tar.gz
```

So I wouldn't rely on it.

scheibelp commented:
https://github.com/spack/spack/pull/10786 is now merged

bebosudo commented:
I think i did a mistake when rebasing on master, and kept also the explicit urls to the last two netcdf versions as well as the new method introduced in #10786, which also updated netcdf.
I removed the explicit urls with commit #2a01cf1, so basically the only change is in the new hdf5 version.
